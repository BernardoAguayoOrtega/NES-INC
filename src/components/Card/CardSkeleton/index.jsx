// import react
import React from 'react';
//import react-loading-skeleton
import Skeleton from 'react-loading-skeleton';

const CardSkeleton = () => {
  return (
    <div className="card" aria-label="card" >
      <Skeleton width={380} height={320}/>
    </div>
  )
};

export default CardSkeleton;
