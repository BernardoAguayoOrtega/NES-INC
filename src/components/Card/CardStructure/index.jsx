// import react
import React from 'react';

const CardStructure = ({ images = [], title }) => {
  const [src] = images;
  const Newsrc = (!src) ? 'https://blog.davidrojo.es/wp-content/uploads/2011/04/file-not-found.jpg' : src;

  return (
    <div className="card" aria-label="card">
      <div className="card__content" aria-label="card content">
        <img
          aria-label="card image"
          className="card__content--image"
          src={`${
            /^https?:\/\/|^http?:\/\//.test(Newsrc) ?
              Newsrc :
              `${/^www?:\/\//.test(Newsrc)}` &&
                `https://${Newsrc}`
          }`}
        />
        <h2 aria-label="image title" className="card__content--title">
          {title}
        </h2>
      </div>
    </div>
  );
};

export default CardStructure;
