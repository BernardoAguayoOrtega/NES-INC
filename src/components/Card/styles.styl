@import "../../stylus/variables.styl";

.card
  width: 100%
  display: inline-block
  font-family: Font1
  border-radius: 5px
  &__content
    position: relative
    z-index: 2
    // background: linear-gradient(0deg, black 50%, transparent)
    background-color: rgba(0,0,0,.8)
    border-radius: 5px
    &:hover
      cursor: pointer
      background: linear-gradient(0deg, #08c, transparent)
      transition: .3s;
    &--image
      width: 100%
      opacity: .7
      border-radius: 5px
    &--title
      z-index: 3
      position: absolute
      bottom: 0
      padding: .9rem
      line-height: 1.1
      font-size: 3.3rem
      font-weight: 800
      color: white
      text-shadow: 2px 3px black;

@media screen and (max-width: 480px)
  .card
    &__content--title
      font-size: 2.8rem
@media screen and (min-width: 481px)
  .card
    &__content--title
      font-size: 3.5rem
@media screen and (min-width: 767px)
  .card
    &__content--title
      font-size: 3.2rem
@media screen and (min-width: 1024px)
  .card
    &__content--title
      font-size: 3.3rem