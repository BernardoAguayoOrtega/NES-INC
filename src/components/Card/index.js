import React from 'react';
import './styles.styl';

const Card = ({ image, title }) => {
  return (
    <div className="card">
      <div className="card__content">
        <img className="card__content--image" src={image}/>
        <h2 className="card__content--title">{title}</h2>
      </div>
    </div>
  )
}

export default Card;
