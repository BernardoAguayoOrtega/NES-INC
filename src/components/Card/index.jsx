// import react
import React, { useState, useEffect } from 'react';
// import card skeleton
import CardStructure from './CardStructure/index.jsx';
// import card skeleton
import CardSkeleton from './CardSkeleton/index.jsx';
// import styles
import './styles.styl';

const Card = ({ image, title }) => {
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 2000);
  }, []);

  return loading ? <CardSkeleton/> : <CardStructure image={image} title={title}/>;
};

export default Card;
