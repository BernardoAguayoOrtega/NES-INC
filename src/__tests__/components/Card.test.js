// import react
import React from 'react';
// import shallow
import { shallow, mount } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import context
import { ContextProvider } from '../../utils/Context/index.jsx';
// import app component
import Card from '../../components/Card/index.jsx';

describe('<Card/> mount', () => {
  const wrapper = mount(
    <ContextProvider>
      <Card images="images" title="title" />
    </ContextProvider>,
  );

  it('is rendering the Card?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is rendering the skeleton?', () => {
    expect(wrapper.find('CardSkeleton').length).toBe(1);
  });

  it('is rendering the skeleton?', () => {
    expect(wrapper.find('loading').exists()).toBe(false);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
