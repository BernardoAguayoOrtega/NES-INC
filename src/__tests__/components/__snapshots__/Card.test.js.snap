// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> matches the snap shot 1`] = `<CardSkeleton />`;
